exonum/src/events/internal.rs

```diff
@@ -71,6 +82,41 @@ impl InternalPart {
                         }).map_err(|_| panic!("Can't execute shutdown"));
                         to_box(f)
                     }
+                    InternalRequest::VerifyTx(tx) => {
+                        let internal_tx = internal_tx.clone();
+                        let tx = Arc::new(tx);
+                        let thread_pool = Rc::clone(&thread_pool);
+
+                        let f = future::loop_fn(Err(SpawnError::at_capacity()), move |status| {
```

Comment:
I didn't like  this retry mechanism. It's unreadable.

can we just  rewrite it as:
```rust
thread_pool.sender().spawn(future::lazy(move || {
                                            if tx.verify() {
                                                internal_tx
                                                    .wait()
                                                    .send(InternalEvent::TxVerified(
                                                        tx.raw().clone(),
                                                    ))
                                                    .expect("Cannot send TxVerified event.");
                                            }
                                            Ok(())
                                        })).expect("cannot verify new transaction);

```

Based on source of tokio-threadpool it will return `SpawnError::at_capacity()` error, only if `num_futures` overflow `MAX_FUTURES` which is `usize::MAX >> 2` probably this is a good value for us, and we really unexpected so many messages. 

@slowli @andreyluchin https://github.com/tokio-rs/tokio/blob/master/tokio-threadpool/src/sender.rs#L99 https://github.com/tokio-rs/tokio/blob/master/tokio-threadpool/src/pool/state.rs#L30

Reply:
I don't want to forgo gracious error handling in favor of readability. To be fair, whole `internal.rs` is unreadable. I would much rather add some comments to make things clearer.